openff/interchange/operations/_combine.py

```diff
@@ -147,6 +151,10 @@ def _combine(
                     _tmp_pot_key.mult = _mult
                     _mult += 1
 
+            if _tmp_pot_key in self_collection.potentials:
+                logging.info(f"Key collision, fixing. Key is {_tmp_pot_key}")
+                _tmp_pot_key.id = _tmp_pot_key.id + "_DUPLICATE"
```

Comment:
(not blocking) This could in theory lead to weird stuff like `c1_DUPLICATE_DUPLICATE_DUPLICATE` after a bunch of component additions. But 1) I don't think that's a big deal, 2) I don't think that's something that will happen often, and 3) I can't really propose a better scheme without a larger structural change.